src/install/bin.zig

```diff
@@ -320,15 +321,19 @@ pub const Bin = extern struct {
             if (!link_global) {
                 target_buf[0..".bin/".len].* = ".bin/".*;
                 from_remain = target_buf[".bin/".len..];
-                dest_buf[0.."../".len].* = "../".*;
-                remain = dest_buf["../".len..];
+                dest_buf[0.."..".len].* = "..".*;
+                remain = dest_buf["..".len..];
+                std.mem.copy(u8, remain, this.package_installed_path);
+                remain = remain[this.package_installed_path.len..];
+                remain[0] = std.fs.path.sep;
+                remain = remain[1..];
             } else {
                 if (this.global_bin_dir.fd >= std.math.maxInt(std.os.fd_t)) {
                     this.err = error.MissingGlobalBinDir;
                     return;
                 }
 
-                @memcpy(&target_buf, this.global_bin_path.ptr, this.global_bin_path.len);
+                std.mem.copy(u8, &target_buf, this.global_bin_path);
```

Comment:
std.mem.copy is slower than memcpy

but memcpy musn't overlap

we probably should have a wrapper that makes this stuff safer honestly

like a

```
bun.concat(dest, &.{"a", "b", "c", "d"})
```

Reply:
Ah okay, I was just going for consistency for all usages within this function - will try and make that helper function now.